### PR TITLE
Formik with Yup validation

### PR DESCRIPTION
**Pull request by J3ffJessie**
- Add Formik dependency, set up Formik as our Form library, add the ability to edit goals as well as change functionality of once our goals list is empty it only displays one Add a Goal button
- Add Yup and set up initial validation of input area, running into some issues will get assistance on it tomorrow
- Modify the Add a goal button to appear at the top of the goals list: also change the .push helper method, to a .insert so that we insert a goal at index 0 at the top of our goals list.

### CODE REVIEW

File: src/pages/GoalsPage.js [hunk 1]
```diff
@@ -1,145 +1,85 @@
...
-    editing: false,
-  },
 
-];
 
-const GoalsPage = ({user}) => {
-  const [goals, setGoals] = useState(defaultGoals);
-  const [formState, setFormState] = useState({goal: '', editedGoal: ''});
-  const {goal, editedGoal} = formState;
+const GoalsPageSchema = Yup.object().shape({
```
Comment (MuddyBootsCode) [inline src/pages/GoalsPage.js:10]: Lets add the validation schema into it's own folder called utils and keep it out of the component logic

File: src/pages/GoalsPage.js [hunk 2]
```diff
@@ -1,145 +1,85 @@
...
+  goals: Yup.string()
+  .min(2, 'Too Short!')
```
Comment (MuddyBootsCode) [inline src/pages/GoalsPage.js:12]: Let's provide some meaningful error message, like "At least 3 characters required" something like that.

**Comment by J3ffJessie** [comment]
Fixed Yup validation to reflect validating an object that contains an array instead of trying to validate an array that was an object. Will test at home and verify working and fix if not correct.

**Comment by J3ffJessie** [comment]
Resolved merge conflict.  Validation works, and displays to the user the error message corresponding to the error.  I will show in class before this PR gets merged for assistance on one aspect that still has me stuck.

**Pull request merged**